### PR TITLE
Plugins 3.0 version matching and single-sourcing

### PR DESCRIPTION
**Pull request by lena-larionova**
### Summary

Bringing the plugins versions up to match 3.0.x and every plugin version in Tyler Ball's [spreadsheet](https://docs.google.com/spreadsheets/d/1rYWO73MtdQUI5n085OYA6VHrtD17cbiMg62ureHO2UQ/edit#gid=0).

Specific changes:

* For the following plugins, I only bumped and matched versions to source files instead of single-sourcing the content. Either the content was too complex to resolve in a short time, or it needed more research:
   * Correlation ID, Datadog, Forward Proxy, JWT Signer, Kafka Log, Kafka Upstream, OpenID Connect, Vault Auth, Zipkin 
* Single-sourced everything else, based on the following criteria:
  * The plugin had no changes in 2.x
  * The plugin had only changed once since 2.x began
  * Rate limiting advanced plugin: special case. Someone made a change in the index file for 3.0, so I had to single-source the plugin to make the versioning work.
  * Apache Openwhisk: special case # 2. We don't handle versioning of this plugin, it's not aligned to Kong Gateway, it's an external package. Changed the formatting to match community plugins (no versions).
  * Upstream TLS: removed entirely. Plugin has been deprecated since 1.3.x.

* Added changelogs for single-sourced plugins whenever relevant
* Deleted all files that are no longer needed

~To do: lots of redirects.~ done

### Reason

https://konghq.atlassian.net/browse/DOCU-2395
https://konghq.atlassian.net/browse/FT-2706

### Testing
Needs careful review to make sure I didn't accidentally break a plugin.
All Kong Inc plugins must have the latest version as 3.0 (except OpenWhisk).

### CODE REVIEW

**Comment by lena-larionova** [comment]
@mheap canonical URLs seem to be breaking for single-sourced plugins. Looks like it's not just in this PR, it's happening for previously updated plugins as well - take a look at https://release-gateway-3-0--kongdocs.netlify.app/hub/kong-inc/ldap-auth-advanced/2.6.x.html for example. For some reason, the smoke test didn't fail until just now though.

I went back through history on both the seo test and the canonical URL generator plugin - seems like the canonicals haven't been working for single-sourced plugins for a while (maybe from the beginning?) but the test never triggered.

If you get bored or have some time, please take a look. Otherwise, we can wait until after 3.0 is out to get this fixed.

**Comment by lena-larionova** [comment]
The remaining Vale errors are coming from the yaml section of one of the plugins, which should be getting ignored: https://github.com/Kong/docs.konghq.com/pull/4331/checks?check_run_id=8025416094

Guessing it's another vale quirk. Ignoring for now.

**Comment by mheap** [comment]
@lena-larionova Canonical URL generation fixed

File: app/_hub/kong-inc/kong-terraform-aws/_index.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ license_type: Apache-2.0
 kong_version_compatibility:
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/kong-terraform-aws/_index.md:13]: This page doesn't have a plugin version or Gateway version dropdown in the netlify preview, not sure if that's intended or not.

File: app/_hub/kong-inc/request-size-limiting/_index.md [hunk 2]
```diff
@@ -79,10 +40,13 @@ params:
...
+      minimum_version: "2.3.x"
       required: true
       default: false
       datatype: boolean
       value_in_examples: false
       description: Set to `true` to ensure a valid `Content-Length` header exists before reading the request body.
 ---
 
+## Changelog
+* Added the `require_content_length` configuration option.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/request-size-limiting/_index.md:52]: Should this have a Gateway version associated with it? Looks like maybe 2.3.x?

File: app/_redirects [hunk 3]
```diff
@@ -133,6 +133,101 @@
...
+/hub/kong-inc/mocking/0.*                          /hub/kong-inc/mocking/
+/hub/kong-inc/mtls-auth/0.*                        /hub/kong-inc/mtls-auth/
+/hub/kong-inc/mtls-auth/1.*                        /hub/kong-inc/mtls-auth/
+/hub/kong-inc/oauth2/0.*                           /hub/kong-inc/oauth2/
+/hub/kong-inc/oauth2/1.*                           /hub/kong-inc/oauth2/
+/hub/kong-inc/oauth2-introspection/0.*             /hub/kong-inc/oauth2-introspection/
+/hub/kong-inc/opa/0.*                              /hub/kong-inc/opa/
+/hub/kong-inc/openid-connect/0.*                   /hub/kong-inc/openid-connect/
+/hub/kong-inc/openid-connect/1.*                   /hub/kong-inc/openid-connect/
+/hub/kong-inc/openwhisk/*                          /hub/kong-inc/openwhisk/
```
Comment (cloudjumpercat) [inline app/_redirects:192]: This one resulted in a 404

**Review by cloudjumpercat — COMMENTED** [review]
Left a few comments with some suggested changes

File: app/_hub/kong-inc/kong-terraform-aws/_index.md (hunk 1 above)
Comment (lena-larionova) [reply app/_hub/kong-inc/kong-terraform-aws/_index.md:13]: Yep, that's intended. If it's a `type: integration` plugin, they never have versions. This is one of the only (maybe the only) plugin like this that's managed by Kong; the rest are all external.

**Review by cloudjumpercat — APPROVED** [review]
LGTM!

**Pull request merged**